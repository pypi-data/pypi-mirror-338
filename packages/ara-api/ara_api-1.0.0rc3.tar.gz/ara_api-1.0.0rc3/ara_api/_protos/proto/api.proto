syntax = "proto3";

service DriverManager {
    rpc GetImuDataRPC (GetRequest) returns (IMUData);
    rpc GetSonarDataRPC (GetRequest) returns (SonarData);
    rpc GetAnalogDataRPC (GetRequest) returns (AnalogData);
    rpc GetAttitudeDataRPC (GetRequest) returns (AttitudeData);
    rpc GetOdometryDataRPC (GetRequest) returns (OdometryData);
    rpc GetOpticalFlowDataRPC (GetRequest) returns (OpticalFlowData);
    rpc GetFlagsDataRPC (GetRequest) returns (FlagsData);
    rpc GetMotorDataRPC (GetRequest) returns (MotorData);

    rpc SendRcDataRPC (RcDataData) returns (StatusData);
}

service NavigationManager {
    rpc TakeOFF (AltitudeSetData) returns (StatusData);
    rpc Land (AltitudeSetData) returns (StatusData);
    rpc Move (PointData) returns (StatusData);

    rpc SetVelocity (VelocityData) returns (StatusData);
    rpc SetAltitude (AltitudeSetData) returns (StatusData);

    rpc SetSettings (SettingsData) returns (StatusData);
    rpc SetZeroOdom (CommandData) returns (StatusData);
}

service VisionManager {
    rpc GetImageData (GetRequest) returns (ImageData);
    rpc GetImageStreamData (GetRequest) returns (stream ImageDataStream);
    rpc GetArucoData (GetRequest) returns (ArucoData);
    rpc GetQRData (GetRequest) returns (QRData);
    rpc GetBlobData (GetRequest) returns (BlobData);

    rpc SetSettings (SettingsData) returns (StatusData);
}

// Status Flags
message StatusData          {
    string status = 1;
}

message CommandData         {
    string cmd = 1;
}

message GetRequest          {
    string req = 1;
}

// Sensor Data
message MotorData           {
    int32 motor_1 = 1;
    int32 motor_2 = 2;
    int32 motor_3 = 3;
    int32 motor_4 = 4;
}

message IMUData             {
    Vector3 gyro = 1;
    Vector3 acc = 2; 
}

message AltitudeSetData     {
    float altitude = 1;
}

message RcDataData          {
    uint32 ail = 1; // roll
    uint32 ele = 2; // pitch
    uint32 thr = 3; // throttle
    uint32 rud = 4; // yaw
    uint32 aux_1 = 5;
    uint32 aux_2 = 6;
    uint32 aux_3 = 7;
    uint32 aux_4 = 8;
}

message AttitudeData        {
    Vector3 orient = 1;
}

message AltitudeData        {
    float altitude = 1;
}

message SonarData           {
    float sonar = 1;
}

message OpticalFlowData     {
    int32 quality = 1;
    float flow_rate_x = 2;
    float flow_rate_y = 3;
    float body_rate_x = 4;
    float body_rate_y = 5;
}

message OdometryData        {
    Vector3 pos = 1;
}

message AnalogData          {
    float voltage = 1;
    int32 mAhdrawn = 2;
    int32 rssi = 3;
    float amperage = 4;
}

// Sub msg types
message PointData           {
    Vector3 point = 1;
}

message VelocityData        {
    Vector3 velocity = 1;
}

message SettingsData        {
    string settings = 1;
}

message Vector3             {
    float x = 1;
    float y = 2;
    float z = 3;
}

message FlagsData           {
    string activeSensors = 1;
    string armingDisableFlags = 2;
    string mode = 3;
}

// Image data
message ImageData {
    uint32 height = 1;
    uint32 weight = 2;
    bytes data = 3;
    float noise = 4;
}

message ImageDataStream {
    bytes data = 1;
}

message ArucoMarker {
    int32 id = 1;
    Vector3 position = 2;
    Vector3 orientation = 3;
}

message QRCode {
    string data = 1;
    Vector3 position = 2;
}

message Blob {
    int32 id = 1;
    Vector3 position = 2;
    float size = 3;
}

message BlobData {
    repeated Blob blobs = 1;
}

message QRData {
    repeated QRCode codes = 1;
}

message ArucoData {
    repeated ArucoMarker markers = 1;
}