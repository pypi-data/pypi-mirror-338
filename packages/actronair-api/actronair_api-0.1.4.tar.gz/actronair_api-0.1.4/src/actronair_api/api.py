"""All Actron Air Cloud API wrappers in python."""

from .auth import AbstractAuth
from .const import (
    ACP_BASE_URL,
    ACP_GET_AC_SYSTEM_STATUS_URL,
    ACP_GET_AC_SYSTEMS_URL,
    ACP_OAUTH2_AUTHORIZE_URL,
    ACP_OAUTH2_TOKEN_URL,
    ACP_POST_CMD_URL,
)
from .exceptions import ApiException, AuthException
from .model import ACSystem, CommandResponse, SystemStatus


def check_status(status) -> any:
    if status == 401:
        raise AuthException(f"Authorization failed: {status}")
    if status != 200:
        raise ApiException(f"Error request failed: {status}")


class ActronAirApi:
    """ActronAir Cloud Platfrom Connection."""

    def __init__(self, auth: AbstractAuth, retryCount: int = 3):
        self.auth = auth
        self.retryCount = retryCount

    async def async_getACSystems(self) -> list[ACSystem]:
        """ActronAir API to fetch all the AC systems linked to the user account."""
        acSystemsDict = {}
        attemptCounter = 0
        error = None
        while attemptCounter < self.retryCount:
            try:
                attemptCounter += 1
                resp = await self.auth.request(
                    "get", ACP_BASE_URL + ACP_GET_AC_SYSTEMS_URL
                )
                check_status(resp.status)
                acSystemsDict = await resp.json(content_type=None)
                return ACSystem.extract_ac_systems(acSystemsDict)
            except () as excep:
                error = excep
        raise ApiException(
            f"Attempt - {attemptCounter}. Failed to fetch AC Systems due to - {error}"
        )

    async def async_getACSystemStatus(self, serial: str) -> any:
        """Actron Air API to fetch AC system status."""
        attemptCounter = 0
        error = None
        while attemptCounter < self.retryCount:
            try:
                attemptCounter += 1
                resp = await self.auth.request(
                    "get",
                    ACP_BASE_URL + ACP_GET_AC_SYSTEM_STATUS_URL.format(WCSerial=serial),
                )
                check_status(resp.status)
                return SystemStatus.extract_system_status(
                    await resp.json(content_type=None)
                )
            except () as excep:
                error = excep
        raise ApiException(
            f"Attempt - {attemptCounter}. Failed to fetch AC System status due to - {error}"
        )

    async def async_sendCommand(self, serial: str, command: str) -> any:
        """Actron Air API to send command to the AC System/Wall Controller."""
        attemptCounter = 0
        error = None
        while attemptCounter < self.retryCount:
            try:
                attemptCounter += 1
                resp = await self.auth.request(
                    "post",
                    ACP_BASE_URL + ACP_POST_CMD_URL.format(WCSerial=serial),
                    json=command,
                )
                check_status(resp.status)
                return CommandResponse.extract_command_response(
                    await resp.json(content_type=None)
                )
            except () as excep:
                error = excep
        raise ApiException(
            f"Attempt - {attemptCounter}. Failed to send command to the AC System due to - {error}"
        )
