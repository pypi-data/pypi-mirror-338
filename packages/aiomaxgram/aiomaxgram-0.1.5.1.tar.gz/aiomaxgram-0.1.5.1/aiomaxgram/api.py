"""
API-клиент для работы с API Max
"""

from typing import Dict, Any, List, Optional, Union
from maxgram.core.network.client import Client

class Api:
    """
    API-клиент для работы с API Max
    
    Класс предоставляет методы для работы с различными ресурсами API.
    """
    
    def __init__(self, token: str, client_options: Optional[Dict[str, Any]] = None):
        """
        Инициализация API-клиента
        
        Args:
            token: Токен доступа бота
            client_options: Дополнительные настройки клиента
        """
        self.client = Client(token, client_options)
    
    async def get_my_info(self) -> Dict[str, Any]:
        """
        Получает информацию о текущем боте
        
        Returns:
            Информация о боте
        """
        return await self.client.request("GET", "/me")
    
    async def set_my_commands(self, commands: List[Dict[str, str]]) -> Dict[str, Any]:
        """
        Устанавливает команды для бота через PATCH /me
        
        Args:
            commands: Список команд в формате [{"name": "command", "description": "Description"}]
            
        Returns:
            Результат запроса PATCH /me
        """
        return await self.client.request("PATCH", "/me", data={"commands": commands})
    
    async def send_message(self, chat_id: int, text: str, attachments: Optional[List[Dict[str, Any]]] = None) -> Dict[str, Any]:
        """
        Отправляет сообщение в чат
        
        Args:
            chat_id: Идентификатор чата
            text: Текст сообщения
            attachments: Вложения сообщения
            
        Returns:
            Информация об отправленном сообщении
        """
        # Параметры запроса
        params = {
            "chat_id": chat_id
        }
        
        # Тело запроса
        data = {
            "text": text
        }
        
        if attachments:
            data["attachments"] = attachments
        
        return await self.client.request("POST", "/messages", params=params, data=data)
    
    async def edit_message(self, message_id: str, text: str, attachments: Optional[List[Dict[str, Any]]] = None) -> Dict[str, Any]:
        """
        Редактирует существующее сообщение
        
        Args:
            message_id: Идентификатор сообщения
            text: Новый текст сообщения
            attachments: Новые вложения сообщения
            
        Returns:
            Информация об отредактированном сообщении
        """
        # Параметры запроса
        params = {
            "message_id": message_id
        }
        
        # Тело запроса
        data = {
            "text": text
        }
        
        if attachments:
            data["attachments"] = attachments
        
        return await self.client.request("PUT", "/messages", params=params, data=data)
    
    async def answer_callback(self, callback_id: str, text: Optional[str] = None) -> Dict[str, Any]:
        """
        Отправляет ответ на колбэк
        
        Args:
            callback_id: Идентификатор колбэка
            text: Текст уведомления, которое увидит пользователь
            
        Returns:
            Результат операции
        """
        # Параметры запроса с callback_id
        params = {
            "callback_id": callback_id
        }
        
        # Тело запроса с уведомлением
        data = {}
        if text:
            data["notification"] = text
        
        return await self.client.request("POST", "/answers", params=params, data=data)
    
    def get_updates(self, allowed_updates: List[str], extra: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """
        Получает новые обновления от API через лонгполлинг
        
        Args:
            allowed_updates: Список типов обновлений, которые нужно получать
            extra: Дополнительные параметры запроса
            
        Returns:
            Список обновлений
        """
        params = extra or {}
        
        if allowed_updates:
            params["types"] = ",".join(allowed_updates)
        
        return self.client.request("GET", "/updates", params=params) 