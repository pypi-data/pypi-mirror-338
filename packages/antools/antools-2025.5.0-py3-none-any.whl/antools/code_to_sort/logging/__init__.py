from ._logger_class import get_logger