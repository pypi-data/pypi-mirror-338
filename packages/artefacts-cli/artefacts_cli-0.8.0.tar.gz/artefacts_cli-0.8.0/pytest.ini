[pytest]
addopts = -ra --cov-config .coveragerc --cov-report term-missing --cov-fail-under 25
console_output_style = progress
env =
    ARTEFACTS_CLI_ENV=test
markers =
  ros2: mark a test as it requires ROS2 environment.
