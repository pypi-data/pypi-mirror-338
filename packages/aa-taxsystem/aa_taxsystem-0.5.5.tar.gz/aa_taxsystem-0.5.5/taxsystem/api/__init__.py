import logging

from ninja import NinjaAPI
from ninja.security import django_auth

from django.conf import settings

from taxsystem.api import taxsystem

logger = logging.getLogger(__name__)

api = NinjaAPI(
    title="Geuthur API",
    version="0.1.0",
    urls_namespace="taxsystem:new_api",
    auth=django_auth,
    csrf=True,
    openapi_url=settings.DEBUG and "/openapi.json" or "",
)

# Add the taxsystem endpoints
taxsystem.setup(api)
