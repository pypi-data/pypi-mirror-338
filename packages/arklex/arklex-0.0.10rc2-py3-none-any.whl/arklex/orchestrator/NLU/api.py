import sys
from pathlib import Path
sys.path.append(str(Path(__file__).resolve().parents[3]))

import logging
import string

from fastapi import FastAPI, Response

from arklex.utils.slot import Verification, SlotInputList, SlotOutputList, SlotOutputListGemini, format_slots, format_slotfilling_input, format_slotfilling_output
from dotenv import load_dotenv
load_dotenv()

from arklex.utils.model_config import MODEL
from arklex.utils.model_provider_config import PROVIDER_MAP
from langchain_openai import ChatOpenAI
from langchain_core.output_parsers.openai_tools import  JsonOutputToolsParser
from pydantic_ai import Agent


logger = logging.getLogger(__name__)


class NLUModelAPI ():
    def __init__(self):
        self.user_prefix = "user"
        self.assistant_prefix = "assistant"

    def get_response(self, sys_prompt, model, response_format="text", note="intent detection"):
        logger.info(f"Prompt for {note}: {sys_prompt}")
        dialog_history = [{"role": "system", "content": sys_prompt}]
        kwargs = {'model': MODEL["model_type_or_path"], 'temperature': 0.7}
        
        if MODEL['llm_provider'] != 'anthropic': kwargs['n'] = 1
        llm = PROVIDER_MAP.get(MODEL['llm_provider'], ChatOpenAI)(**kwargs)

        if MODEL['llm_provider'] == 'openai':
            llm = llm.bind(response_format={"type": "json_object"} if response_format == "json" else {"type": "text"})
            res = llm.invoke(dialog_history)
        else:
            messages = [("user", f"{dialog_history[0]['content']} Only choose the option letter, no explanation.")]
            res = llm.invoke(messages)
        
        return res.content

    def format_input(self, intents, chat_history_str) -> str:
        """Format input text before feeding it to the model."""
        intents_choice, definition_str, exemplars_str = "", "", ""
        idx2intents_mapping = {}
        multiple_choice_index = dict(enumerate(string.ascii_lowercase))
        count = 0
        for intent_k, intent_v in intents.items():
            if len(intent_v) == 1:
                intent_name = intent_k
                idx2intents_mapping[multiple_choice_index[count]] = intent_name
                definition = intent_v[0].get("attribute", {}).get("definition", "")
                sample_utterances = intent_v[0].get("attribute", {}).get("sample_utterances", [])

                if definition:
                    definition_str += (
                        f"{multiple_choice_index[count]}) {intent_name}: {definition}\n"
                    )
                if sample_utterances:
                    exemplars = "\n".join(sample_utterances)
                    exemplars_str += (
                        f"{multiple_choice_index[count]}) {intent_name}: \n{exemplars}\n"
                    )
                intents_choice += f"{multiple_choice_index[count]}) {intent_name}\n"

                count += 1

            else:
                for idx, intent in enumerate(intent_v):
                    intent_name = f'{intent_k}__<{idx}>'
                    idx2intents_mapping[multiple_choice_index[count]] = intent_name
                    definition = intent.get("attribute", {}).get("definition", "")
                    sample_utterances = intent.get("attribute", {}).get("sample_utterances", [])

                    if definition:
                        definition_str += (
                            f"{multiple_choice_index[count]}) {intent_name}: {definition}\n"
                        )
                    if sample_utterances:
                        exemplars = "\n".join(sample_utterances)
                        exemplars_str += (
                            f"{multiple_choice_index[count]}) {intent_name}: \n{exemplars}\n"
                        )
                    intents_choice += f"{multiple_choice_index[count]}) {intent_name}\n"

                    count += 1
        # Base prompt without conditional sections
        system_prompt_nlu = """According to the conversation, decide what is the user's intent in the last turn? \n"""
        
        # Conditionally add definitions if available
        if definition_str.strip():
            system_prompt_nlu += """Here are the definitions for each intent:\n{definition}\n"""
            
        # Conditionally add exemplars if available
        if exemplars_str.strip():
            system_prompt_nlu += """Here are some sample utterances from user that indicate each intent:\n{exemplars}\n"""
            
        # Add the rest of the prompt
        system_prompt_nlu += """Conversation:\n{formatted_chat}\n\nOnly choose from the following options.\n{intents_choice}\n\nAnswer:"""
        
        system_prompt = system_prompt_nlu.format(
            definition=definition_str,
            exemplars=exemplars_str,
            intents_choice=intents_choice,
            formatted_chat=chat_history_str,
        )
        return system_prompt, idx2intents_mapping

    def predict(
        self,
        text,
        intents,
        chat_history_str,
        model
    ) -> str:

        system_prompt, idx2intents_mapping = self.format_input(
            intents, chat_history_str
        )
        response = self.get_response(
            system_prompt, model, note="intent detection"
        )
        logger.info(f"postprocessed intent response: {response}")
        try:
            pred_intent_idx = response.split(")")[0]
            pred_intent = idx2intents_mapping[pred_intent_idx]
        except:
            pred_intent = response.strip().lower()
        logger.info(f"postprocessed intent response: {pred_intent}")
        return pred_intent


class SlotFillModelAPI():
    def __init__(self):
        self.user_prefix = "user"
        self.assistant_prefix = "assistant"
    
    # System prompt for slot filling
    def format_input(self, input_slots: SlotInputList, chat_history_str) -> str:
        """Format input text before feeding it to the model."""
        system_prompt = f"Given the conversation and definition of each dialog state, update the value of following dialogue states.\nDialogue Statues:\n{input_slots}\nConversation:\n{chat_history_str}\n\n"
        return system_prompt

    # get response from model
    def get_response(self, sys_prompt, format=SlotOutputList, note="slot filling"):
        logger.info(f"Prompt for {note}: {sys_prompt}")
        dialog_history = [{"role": "system", "content": sys_prompt}]
        kwargs = {'model': MODEL["model_type_or_path"], 'temperature': 0.7}
        # set number of chat completions to generate, isn't supported by Anthropic
        if MODEL['llm_provider'] != 'anthropic': kwargs['n'] = 1
        llm = PROVIDER_MAP.get(MODEL['llm_provider'], ChatOpenAI)(**kwargs)
        
        if MODEL['llm_provider'] == 'openai':
            llm = llm.with_structured_output(schema=format)
            response = llm.invoke(dialog_history)
    
        elif MODEL['llm_provider']=='huggingface':
            llm = llm.bind_tools([format])
            chain =  llm | JsonOutputToolsParser()
            res = chain.invoke(dialog_history)

        elif MODEL['llm_provider'] == 'gemini':
            agent = Agent(f"google-gla:{MODEL['model_type_or_path']}", result_type=SlotOutputListGemini)
            result = agent.run_sync(dialog_history[0]['content'])
            response = result.data

        #for claude 
        else:
            messages = [{"role": "user", "content": dialog_history[0]['content']}]
            llm = llm.bind_tools([format])
            res= llm.invoke(messages)
            response = format(**res.tool_calls[0]['args'])
        return response

    # endpoint for slot filling
    def predict(
        self,
        slots,
        chat_history_str,
    ):
        formatted_slots = format_slots(slots)
        input_slots = format_slotfilling_input(formatted_slots)
        system_prompt = self.format_input(input_slots, chat_history_str)
        response = self.get_response(system_prompt, note="slot filling")
        filled_slots = format_slotfilling_output(formatted_slots, response)
        logger.info(f"Updated dialogue states: {filled_slots}")
        return filled_slots
    
    # endpoint for slot verification
    def verify(
        self,
        slot: dict,
        chat_history_str,
    ) -> Verification:
        reformat_slot = {key: value for key, value in slot.items() if key in ["name", "type", "value", "enum", "description", "required"]}
        system_prompt = f"Given the conversation, definition and extracted value of each dialog state, decide whether the following dialog states values need further verification from the user. Verification is needed for expressions which may cause confusion. If it is an accurate information extracted, no verification is needed. If there is a list of enum value, which means the value has to be chosen from the enum list. Only Return boolean value: True or False. \nDialogue Statues:\n{reformat_slot}\nConversation:\n{chat_history_str}\n\n"
        response = self.get_response(
            system_prompt, format=Verification, note="slot verification"
        )
        if not response: # no need to verification, we want to make sure it is really confident that we need to ask the question again
            logger.info(f"Failed to verify dialogue states")
            return Verification(verification_needed=False, thought="No need to verify")
        logger.info(f"Verified dialogue states: {response}")
        return response


app = FastAPI()
nlu_api = NLUModelAPI()
slotfilling_api = SlotFillModelAPI()


@app.post("/nlu/predict")
def predict(data: dict, res: Response):
    logger.info(f"Received data: {data}")
    pred_intent = nlu_api.predict(**data)

    logger.info(f"pred_intent: {pred_intent}")
    return {"intent": pred_intent}

@app.post("/slotfill/predict")
def predict(data: dict, res: Response):
    logger.info(f"Received data: {data}")
    results = slotfilling_api.predict(**data)
    logger.info(f"pred_slots: {results}")
    return results

@app.post("/slotfill/verify")
def verify(data: dict, res: Response):
    logger.info(f"Received data: {data}")
    verify_needed = slotfilling_api.verify(**data)

    logger.info(f"verify_needed: {verify_needed}")
    return verify_needed