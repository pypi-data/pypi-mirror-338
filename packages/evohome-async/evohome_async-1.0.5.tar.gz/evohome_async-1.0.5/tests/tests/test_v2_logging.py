"""Tests for evohome-async - validate the schemas of vendor's RESTful JSON."""

from __future__ import annotations

import logging
from pathlib import Path
from typing import TYPE_CHECKING
from unittest.mock import patch

from tests.conftest import EvohomeClientv2

from .conftest import FIXTURES_V2, auth_get

if TYPE_CHECKING:
    import pytest
    from freezegun.api import FrozenDateTimeFactory
    from syrupy import SnapshotAssertion

    from tests.conftest import CredentialsManager


def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
    folders = [
        p
        for p in Path(FIXTURES_V2).glob("*")
        if p.is_dir() and not p.name.startswith("_")
    ]
    metafunc.parametrize(
        "fixture_folder", sorted(folders), ids=(p.name for p in sorted(folders))
    )


async def test_system_warnings(
    credentials_manager: CredentialsManager,
    fixture_folder: Path,
    freezer: FrozenDateTimeFactory,
    caplog: pytest.LogCaptureFixture,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the user account schema against the corresponding JSON."""

    freezer.move_to("2025-01-01T00:00:00+00:00")

    with patch("evohomeasync2.auth.Auth.get", auth_get(fixture_folder)):
        evo = EvohomeClientv2(credentials_manager)

        with caplog.at_level(logging.WARNING):
            await evo.update()

    assert caplog.record_tuples == snapshot
