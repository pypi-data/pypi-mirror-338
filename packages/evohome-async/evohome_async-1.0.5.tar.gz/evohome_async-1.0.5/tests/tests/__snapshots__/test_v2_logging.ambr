# serializer version: 1
# name: test_system_warnings[default]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432521'): Active fault: 2024-12-17T17:10:57+00:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432576'): Active fault: 2024-05-01T01:00:00+01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[hass_118169]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432521'): Active fault: 2024-12-17T17:10:57+00:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432576'): Active fault: 2024-05-01T01:00:00+01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[hass_139906]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432521'): Active fault: 2024-12-17T17:10:57+00:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432576'): Active fault: 2024-05-01T01:00:00+01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[hass_139945]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432521'): Active fault: 2024-12-17T17:10:57+00:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432576'): Active fault: 2024-05-01T01:00:00+01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[hass_140194]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432521'): Active fault: 2024-12-17T17:10:57+00:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='3432576'): Active fault: 2024-05-01T01:00:00+01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[system_002]
  list([
    tuple(
      'evohomeasync2',
      30,
      "HotWater(id='3933910'): Active fault: 2023-11-30T18:48:40+00:00 DHWSensorCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[system_004]
  list([
    tuple(
      'evohomeasync2',
      30,
      'There are 4 locations. Reduce the risk of exceeding API rate limits by individually updating only necessary locations.',
    ),
    tuple(
      'evohomeasync2',
      30,
      "Gateway(id='2938388'): Active fault: 2023-05-04T18:47:36.772704+02:00 GatewayCommunicationLost",
    ),
  ])
# ---
# name: test_system_warnings[system_006]
  list([
    tuple(
      'evohomeasync2',
      30,
      "Unable to find IANA TZ identifier for 'NewYork'; DST support will be very limited",
    ),
    tuple(
      'evohomeasync2',
      30,
      'There are 2 locations. Reduce the risk of exceeding API rate limits by individually updating only necessary locations.',
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='0004'): Active fault: 2024-12-01T01:00:00+01:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='0004'): Active fault: 2024-05-01T01:00:00+02:00 TempZoneSensorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='0004'): Active fault: 2024-12-01T01:00:00-01:00 TempZoneActuatorCommunicationLost",
    ),
    tuple(
      'evohomeasync2',
      30,
      "Zone(id='0004'): Active fault: 2024-05-01T01:00:00-01:00 TempZoneSensorCommunicationLost",
    ),
  ])
# ---
